addons/portal_rating/static/src/js/portal_rating_composer.js

```diff
@@ -56,17 +56,56 @@ var RatingPopupComposer = publicWidget.Widget.extend({
 
 publicWidget.registry.RatingPopupComposer = publicWidget.Widget.extend({
     selector: '.o_rating_popup_composer',
+    custom_events: {
+        reload_rating_popup_composer: '_onReloadRatingPopupComposer',
+    },
 
     /**
      * @override
      */
     start: function () {
-        var ratingPopupData = this.$el.data();
-        var ratingPopup = new RatingPopupComposer(this, ratingPopupData);
+        this.ratingPopupData = this.$el.data();
+        this.ratingPopupData.display_composer = this.ratingPopupData.disable_composer ? false : !session.is_website_user;
+        this.ratingPopup = new RatingPopupComposer(this, this.ratingPopupData);
         return Promise.all([
             this._super.apply(this, arguments),
-            ratingPopup.appendTo(this.$el)
+            this.ratingPopup.appendTo(this.$el)
         ]);
     },
+
+    //--------------------------------------------------------------------------
+    // Private
+    //--------------------------------------------------------------------------
+
+    /**
+     * Destroy existing ratingPopup and insert new ratingPopup widget
+     * @private
+     * @param {Object} data
+     */
+    _ratingPopupComposerWidget: function (data) {
+        // destroy existing ratingPopup
+        var oldComposer = this.ratingPopup;
+        if (oldComposer) {
+            oldComposer.destroy();
+        }
```

Comment:
Why are we creating an intermediary variable here instead of simply:

```javascript
if (this.ratingPopup) {
    this.ratingPopup.destroy();
}
```

Reply:
Done! 👍